### ISSUE TITLE
Working with multiple datasources

### ISSUE THREAD

**Issue by pkafel** [flyway/flyway-test-extensions#25]
Hey all !

I would like to propose a feature - support for multiple datasources. You could easily (at least I think so) achieve that by extending `@FlywayTest` annotation and `FlywayTestExecutionListener` class. This could look like this:

```java
@Retention(RetentionPolicy.RUNTIME)
public @interface FlywayTest {
  // current implementation goes here
  public String datasourceName() default "";
  // more current implementation goes here
}
```

In `FlywayTestExecutionListener` when reading from Spring context you are just picking correct datasource. I put default to empty String so :
* this change will not break any current code
* empty string could be an indicator that default datasource should be used

Hope you like the idea !

**Comment by FlorianGWE** [comment]
I see this feature will not break any current code.

I only want better understand, why you need in a test case a specific datasoure. With the annotation the datasource is hard coded and can not be changed. 

In my opinion you will get a simular solution with a configuration Class and a profile configuration. 
 see https://docs.spring.io/spring/docs/current/spring-framework-reference/html/integration-testing.html. (Spring 4)
I had not tried, because I currently have no use case for change the datasource for specific Test Class or a specific Test in a Test Class.

**Comment by pkafel** [comment]
Hey Florian,

I went quickly through the code and now I see from where does your concerns come from. So my idea was based on the fact that Spring is not really good in working with multiple datasources and flyway in the same moment. When you have more than one datasource all of the sudden you need to handle everything on your own because you have two flyways migrations, two sets of properties etc. So nothing works out of the box. Especially not integration tests.

Now I see that this will not be as easy as I thought when I was creating ticket. I see that you have assumption that there is no or exactly one `Flyway` instance in Spring context which means that in order to support multiple datasources you would need to rewrite the whole logic a bit...

**Comment by FlorianGWE** [comment]
Multible Datasource will work, but not out of the box. 
In this case you must do a little bit more configuration, but you can it access from Spring and also with Flyway. 

For a better understanding what do you understand under `different datasource`?

Do you mean:
- different database instances or vendor
- different schema in the same database instance.

If you can me explain a little bit your scenario than may be I got a idea for good solution. 
May be only with spring configuration or add a little bit code to Flyway Test to made this possible.

**Comment by pkafel** [comment]
By different datasources I mean different database instances or vendor. As a corner case of course if you point them to the same database with just different schema this should work as well.

**Comment by FlorianGWE** [comment]
How do you want setup your test?

1. Run in one test class different test agains different databases?
1. Run each test class and test again different databases?

For 1. I think we need for each @Flyway annotation a different Flyway instances with the database configuration. We must extent the runnert so the @Flyway annotation can reference the id/name of the needed Flyway instance.
For 2. this can be done with Maven and Spring configuration and using profiles.

**Comment by FlorianGWE** [comment]
I have implement possibility to use a different Flyway configuration in test.
I feel better with this solulion as change the database connection of Flyway secretly.

Please take a look in annotation implementation @[FlywayTest](https://github.com/flyway/flyway-test-extensions/blob/master/flyway-test-extensions/flyway-spring-test/src/main/java/org/flywaydb/test/annotation/FlywayTest.java) and also in a example Test [SpringConfigTest ](https://github.com/flyway/flyway-test-extensions/blob/master/flyway-test-extensions/flyway-test-samples/flyway-test-spring-samples/flyway-test-sample-spring4/src/test/java/org/flywaydb/test/sample/spring4/SpringConfigTest.java ).

Example usage:
`@FlywayTest(flywayName="flyway-instance-name")`

**Comment by FlorianGWE** [comment]
My fix release with version 4.0.1
